src/main/webapp/calendar-script.js

```diff
@@ -43,8 +66,95 @@ function getEvents(calendar) {
         title: event.name,
         start: event.strStartTime,
         end: event.strEndTime,
-        allDay: false
+        allDay: false,
+        extendedProps: {
+          stringDate: event.strStartTime,
+          address: event.address,
+          placeId: event.placeId, 
+          openTime: '9AM',
+          closeTime: '5PM',
+        }
       });
     });
   });
 }
+
+function createMap(modalBody, eventObj) {
+  const infoDisplay = document.createElement('div');
+  let address = document.createElement('p');
+  var urlFormatAddress = eventObj.extendedProps.address.replace(/\s/g, '+')
+  var url = 'https://www.google.com/maps/search/?api=1&query=' + urlFormatAddress + '&query_place_id=' + eventObj.extendedProps.placeId;
+  infoDisplay.innerHTML = '<b>Address: </b><a href=' + url + '>' + eventObj.extendedProps.address + '</a>' + '<br>';
+  infoDisplay.appendChild(address);
+  modalBody.appendChild(infoDisplay)
+
+  // create new div to hold map
+  const mapDis = document.createElement('div');
+  mapDis.id = 'map';
+  modalBody.appendChild(mapDis);
+
+  // instantiate map
+  map = new google.maps.Map(document.getElementById('map'), {
+      zoom: 13,
+      center: new google.maps.LatLng(0, 0)
+    });
+
+  service = new google.maps.places.PlacesService(map);
+  service.getDetails({
+    placeId: eventObj.extendedProps.placeId
+  }, function(result, status) {
+    if (status != google.maps.places.PlacesServiceStatus.OK) {
+      alert(status);
+      return;
+    }
+    // set center of map
+    map.setCenter(result.geometry.location);
+  
+    // add open hours to mod info Display
+    let openHours = document.createElement('p');
+
+    // if there are open hours display them, otherwise open all day
+    try {
+      var stringFullDate = eventObj.extendedProps.stringDate;
+      var dateStr = stringFullDate.split('T')[0]; 
+      var intOfWeek = getintOfWeek(dateStr);
+
+      var dateObj = new Date(stringFullDate);
+      var openTime = formatAMPM(result.opening_hours.periods[intOfWeek].open.time);
+      var closeTime = formatAMPM(result.opening_hours.periods[intOfWeek].close.time);
```

Comment:
I know this is late as we already have code in the database that doesn't follow this rule, but the JavaScript style guide notes to use `let` or `const` instead of `var`: https://google.github.io/styleguide/jsguide.html#features-use-const-and-let. This is possibly a nit, I'll let you decide what to do here.

Reply:
There are other `var`s in your codebase, but I won't leave this comment on the rest of them.

Reply:
removed them all